enums.rb

```diff
@@ -0,0 +1,128 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  # rubocop:enable Metrics/ModuleLength
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |i|
+      yield to_a[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |i|
+      yield to_a[i], i
+    end
+    self
+  end
+
+  def my_all?(arg = nil)
+    if block_given? && !arg.nil?
+      return false unless my_all?(arg) && my_all?
+    end
+    if block_given?
+      my_each { |i| return false unless yield(i) }
+    elsif arg.nil?
+      my_each { |i| return false unless i }
+    elsif arg.class == Class
+      my_each { |i| return false unless i.class <= arg }
+    elsif arg.class == Regexp
+      my_each { |i| return false unless i =~ arg }
+    else
+      my_each { |i| return false unless i == arg && i.class <= arg.class }
+    end
+
+    true
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each do |i|
+      new_array.push(i) if yield(i)
+    end
+    new_array
+  end
+
+  def my_any?(arg = nil)
+    if block_given? && !arg.nil?
+      return true unless my_any?(arg) && my_any?
+    end
+
+    if block_given?
+      my_each { |i| return true unless yield(i) }
+    elsif arg.nil?
+      my_each { |i| return true unless i }
+    elsif arg.class == Class
+      my_each { |i| return true unless i.class <= arg }
+    elsif arg.class == Regexp
+      my_each { |i| return true unless i =~ arg }
+    else
+      my_each { |i| return true unless i == arg && i.class <= arg.class }
+    end
+
+    false
+  end
+
+  def my_none?(arg = nil)
+    my_each do |i|
+      if block_given?
+        return false if yield(i)
+      elsif arg.class == Regexp
+        my_each { return true unless i =~ arg }
+      elsif !i.nil? && i != false
+      else
+        return false
+      end
+    end
+    true
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?

- [ ] when no block or argument is given returns true only if none of the collection members is true
```ruby
[nil, false].my_none? == [nil, false].none? => true
```
- Yours is retrieving false (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/86543156-3e434c80-bed9-11ea-89aa-ce0aa5e25b38.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/86543177-5dda7500-bed9-11ea-9513-dd988e211f48.png)

```ruby
[5, -1, true].my_none? == [5, -1, true].none? => false
```
- Yours is retrieving true (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/86543206-a1cd7a00-bed9-11ea-94ae-2e4fa61d3c3d.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/86543209-b0b42c80-bed9-11ea-9746-bf530b6e039c.png)

- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
['1', '2.5', 'a', 9].my_none?(Integer) == ['1', '2.5', 'a', 9].none?(Integer) => false
```
- Yours is retrieving true (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/86543420-86fc0500-bedb-11ea-86f8-82dc7fa18492.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/86543426-9713e480-bedb-11ea-9ae4-6c5195096eea.png)

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rodz blade].my_none?(/z/) == %w[dog door rodz blade].none?(/z/) => false
```
- Yours is retrieving true (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/86543449-c9254680-bedb-11ea-9068-09ae3862c95d.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/86543453-d4787200-bedb-11ea-9de1-2020fad18e64.png)

- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
[3,4,7,11].my_none?(3) == [3,4,7,11].none?(3) => false
```
- Yours is retrieving true (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/86543510-9465bf00-bedc-11ea-8835-f5f6a19c939d.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690915-a9d85e00-6f69-11ea-81b3-7adb07b281ca.png)

[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end

To make this work we have to use block.call(element) instead of yield(element) in my_any?